tests/memories_test.py

```diff
@@ -1437,6 +1437,33 @@ def f(x):
                             out_specs=P('x', 'y')))(arr)
     self.assertArraysEqual(out, np_inp * 24)
 
+  def test_stream_annotation_inside_shmap(self):
```

Comment:
You need to condition this test on xla_extension_version and skip i.e. `if xla_extension_version < ...: self.skipTest(...)`.

See the pattern in some other tests.

Reply:
Also you need to only run this test on GPU. So add a condition for jtu.test_device_matches and skip if not GPU.

Reply:
I added the two checks you suggested. I made the check for the `xla_extension_version` to be 1000 since this feature isn't released yet. I'll change this once it is fully upstreamed.

Reply:
That means that this test won't even run at HEAD. It will be skipped.

You should use 299 I think.

Reply:
If the feature is not ready to be tested in JAX, then the JAX changes shouldn't be made. If this current version works, then 299 should work.